homeassistant/components/crownstone/config_flow.py

```diff
@@ -0,0 +1,296 @@
+                user_input[CONF_USB_SPHERE_OPTION] = sphere_id
+                self.updated_entry_data[CONF_USB_SPHERE] = sphere_id
+
+            self.entry.data = MappingProxyType(self.updated_entry_data)
+            return self.async_create_entry(title="", data=user_input)
+
+        return self.async_show_form(step_id="init", data_schema=options_schema)
+
+    async def async_step_usb_config_option(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Set up a Crownstone USB dongle."""
+        list_of_ports = await self.hass.async_add_executor_job(
+            serial.tools.list_ports.comports
+        )
+        ports_as_string = list_ports_as_str(list_of_ports, False)
+
+        if user_input is not None:
+            selection = user_input[CONF_USB_PATH]
+
+            if selection == MANUAL_PATH:
+                return await self.async_step_usb_manual_config_option()
+            if selection != REFRESH_LIST:
+                selected_port: ListPortInfo = list_of_ports[
+                    ports_as_string.index(selection)
+                ]
+                usb_path = await self.hass.async_add_executor_job(
+                    usb.get_serial_by_id, selected_port.device
+                )
+                self.updated_entry_data[CONF_USB_PATH] = usb_path
+                return await self.async_step_usb_sphere_config_option()
+
+        return self.async_show_form(
+            step_id="usb_config_option",
+            data_schema=vol.Schema(
+                {vol.Required(CONF_USB_PATH): vol.In(ports_as_string)}
+            ),
+        )
+
+    async def async_step_usb_manual_config_option(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Manually enter Crownstone USB dongle path."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="usb_manual_config_option",
+                data_schema=vol.Schema({vol.Required(CONF_USB_MANUAL_PATH): str}),
+            )
+
+        self.updated_entry_data[CONF_USB_PATH] = user_input[CONF_USB_MANUAL_PATH]
+        return await self.async_step_usb_sphere_config_option()
+
+    async def async_step_usb_sphere_config_option(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Select a Crownstone sphere that the USB operates in."""
+        # no need to select if there's only 1 option
+        if len(self.spheres) == 1:
+            user_input = {CONF_USB_SPHERE: next(iter(self.spheres.keys()))}
+
+        if user_input is None:
+            return self.async_show_form(
+                step_id="usb_sphere_config_option",
+                data_schema=vol.Schema({CONF_USB_SPHERE: vol.In(self.spheres.keys())}),
+            )
+
+        self.updated_entry_data[CONF_USB_SPHERE] = self.spheres[
+            user_input[CONF_USB_SPHERE]
+        ]
+        self.entry.data = MappingProxyType(self.updated_entry_data)
```

Comment:
Don't set this directly, let `async_create_entry` set it.

Reply:
When the flow finishes <https://github.com/home-assistant/core/blob/dev/homeassistant/config_entries.py#L1447> will take care of updating the options.  If you have options that will change that are configured via the options flow and the original config flow, you can store them in `options` from the start using https://github.com/home-assistant/core/blob/dev/homeassistant/config_entries.py#L1393